x-pack/plugin/ilm/qa/multi-node/build.gradle

```diff
@@ -2,6 +2,8 @@ apply plugin: 'elasticsearch.java-rest-test'
 
 dependencies {
   javaRestTestImplementation project(path: xpackProject('plugin').path, configuration: 'testArtifacts')
+  // let the javaRestTest see the classpath of main
+  javaRestTestImplementation project.sourceSets.main.runtimeClasspath
```

Comment:
This might be why intellij gets confused. Instead we should add `sourcets.main.runtimeClasspath` to `sourcesets.javaRestTest.runtimeClasspath` rather than wire this up as a filecollection dependency. Basically, we should use `GradleUtils.extendSourceSet()` here if we need `javaRestTest` to "extend from" `main`.

Reply:
relevant prior comment: https://github.com/elastic/elasticsearch/pull/59444/files/c89b5de16f777a00a7be76280565debd6014728f#diff-3f6ea6fd97c3537cc14d8d9f622aac7b

Are there any concerns with using GradleUtils here (and about 5 other places) ? Concerns with example project different from rest of the projects?

Reply:
Yes, since example projects are meant for external plugin authors to use as a baseline for their own projects. Internal use is not so much an issue.

Reply:
done https://github.com/elastic/elasticsearch/pull/62258/commits/87c2493a517df0373e2a94331464e385e5e79db8